technico-v16/app/users/[id]/edit-user/page.tsx

```diff
@@ -0,0 +1,239 @@
+      setFormData(user);
+    } catch (err: any) {
+      setErrorMessage(err.message || "An error occurred while fetching user data.");
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    setFormData((prev) => ({
+      ...prev,
+      [name]: value,
+    }));
+    setErrors((prev) => ({
+      ...prev,
+      [name]: "",
+    }));
+  };
+
+  const validateForm = (): boolean => {
+    const newErrors: FormErrors = {};
+
+    if (!formData.name) newErrors.name = "Name is required.";
+    if (!formData.surname) newErrors.surname = "Surname is required.";
+    if (!formData.vatNumber || !/^\d{9}$/.test(formData.vatNumber)) {
+        newErrors.vatNumber = "VAT Number must be 9 digits.";
+    }
+    if (!formData.address) newErrors.address = "Address is required.";
+    if (!formData.phoneNumber || !/^\d{10}$/.test(formData.phoneNumber)) {
+        newErrors.phoneNumber = "Phone Number must be 10 digits.";
+    }
+    if (!formData.email || !/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(formData.email)) {
+        newErrors.email = "Invalid email address.";
+    }
+    if (!formData.password || formData.password.length < 6) {
+        newErrors.password = "Password must be at least 6 characters.";
+    }
+
+    setErrors(newErrors);
+    return Object.keys(newErrors).length === 0;
+};
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (!validateForm()) {
+      return;
+    }
+
+    try {
+      const response = await fetch(`https://localhost:7118/api/Users/${id}`, {
+        method: "PUT",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(formData),
+      });
+
+      if (!response.ok) {
+        throw new Error(`Error ${response.status}: Failed to update user`);
+      }
+
+      setSuccessMessage("User updated successfully!");
+      setTimeout(() => {
+        router.back();
+      }, 2000);
+    } catch (err: any) {
+      setErrorMessage(err.message || "An error occurred while updating the user.");
+    }
+  };
+
+  if (loading) return <p>Loading...</p>;
+  if (errorMessage) return <p className="text-red-500">{errorMessage}</p>;
+
+  return (
+    <div>
+      <h1 className="text-center text-4xl text-gray-600 m-5">Edit User</h1>
+      <form onSubmit={handleSubmit}>
+        <div className="flex flex-col items-center">
+        <div className="flex items-center w-80">
+          <label className="w-32" htmlFor="name">Name:</label>
+          <input
+            type="text"
+            name="name"
+            placeholder="Name"
+            className={`input input-bordered w-full max-w-xs m-2 ${errors.name && "border-red-500"}`}
+            value={formData.name}
+            onChange={handleChange}
+          />
+          </div>
+          {errors.name && <span className="text-red-500 text-sm">{errors.name}</span>}
```

Comment:
Good usage of the errors object , showing errors related to the inputs and styling them properly